src/main/java/org/spongepowered/api/item/data/ItemBlockData.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.data;
+
+import org.spongepowered.api.block.BlockType;
+import org.spongepowered.api.block.data.TileEntity;
+
+import com.google.common.base.Optional;
+
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+/**
+ * 
+ * Represents data associated with all items that place a block.
+ *
+ */
+public interface ItemBlockData extends ItemData {
+
+    /**
+     * Gets the blocks this block is placeable on. Optional.absent() represents
+     * all blocks.
+     * 
+     * @return The blocks this block is placeable on
+     */
+    Optional<Set<BlockType>> getPlaceableOn();
```

Comment:
@gabizou I don't see why. Optional.absent() indicates the attribute of placeables is absent.